### PR TITLE
Resubmission of #50929 with fixes

### PR DESCRIPTION
**Pull request by fxierh**
#50929 was reverted by #51372 due to the fact that `HC_ZONES` was mistakenly written as `HC_ZONEs` in the following section:
https://github.com/openshift/release/blob/33f416a519108e3cb0a82a089ef4874847964eed/ci-operator/step-registry/hypershift/aws/create/hypershift-aws-create-chain.yaml#L59-L62
causing `periodic-ci-openshift-hypershift-release-4.15-periodics-e2e-aws-ovn-conformance` to fail in the following way:
```bash
/bin/bash: line 30: HC_ZONES: unbound variable 
```

This PR re-submits #50929 with fixes. 

Rehearsal results:
[periodic-ci-openshift-openshift-tests-private-release-4.16-amd64-nightly-aws-ipi-ovn-hypershift-private-mgmt-f7](https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/51393/rehearse-51393-periodic-ci-openshift-openshift-tests-private-release-4.16-amd64-nightly-aws-ipi-ovn-hypershift-private-mgmt-f7/1783816249373888512): ok
[periodic-ci-openshift-hypershift-release-4.15-periodics-e2e-aws-ovn-conformance](https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift_release/51393/rehearse-51393-periodic-ci-openshift-hypershift-release-4.15-periodics-e2e-aws-ovn-conformance/1783764611598651392): ok
[periodic-ci-openshift-openshift-tests-private-release-4.16-amd64-nightly-aws-ipi-ovn-hypershift-private-guest-f7](https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/51393/rehearse-51393-periodic-ci-openshift-openshift-tests-private-release-4.16-amd64-nightly-aws-ipi-ovn-hypershift-private-guest-f7/1783816251890470912): ok (step phase pre & post passed)
[periodic-ci-openshift-hypershift-release-4.16-periodics-e2e-aws-ovn-conformance-serial](https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift_release/51393/rehearse-51393-periodic-ci-openshift-hypershift-release-4.16-periodics-e2e-aws-ovn-conformance-serial/1783764614098456576): ok (step phase pre & post passed)
[periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-aws-ovn-proxy-conformance](https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift_release/51393/rehearse-51393-periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-aws-ovn-proxy-conformance/1783817480674742272): failed due to an **unrelated** error ([this job](https://prow.ci.openshift.org/view/gs/test-platform-results/logs/periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-aws-ovn-proxy-conformance/1784417577695449088) also failed in the same way)

### CODE REVIEW

**Comment by fxierh** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-amd64-nightly-aws-ipi-ovn-hypershift-private-mgmt-f7 periodic-ci-openshift-openshift-tests-private-release-4.16-amd64-nightly-aws-ipi-ovn-hypershift-private-guest-f7

**Comment by fxierh** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-aws-ovn-proxy-conformance

**Comment by fxierh** [comment]
/cc @csrwng
/cc @jianlinliu
/cc @liangxia
/cc @heliubj18

Could you please review the sections you are responsible for in the resubmitted PR ? Thank you.
(Sorry for tagging you for the same thing again)

**Comment by fxierh** [comment]
/cc @jupierce

This PR has been thoroughly rehearsed and should be okay, but could you please take a look to ensure the issues have been resolved appropriately? Thank you.

**Comment by liangxia** [comment]
One of the required job failed
```
ci/prow/core-valid — Job failed.  
```

**Comment by fxierh** [comment]
Just a rebase.

**Comment by fxierh** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.15-periodics-e2e-aws-ovn-conformance periodic-ci-openshift-hypershift-release-4.16-periodics-e2e-aws-ovn-conformance-serial

For safety.

**Comment by fxierh** [comment]
Latest rehearsal results:
[periodic-ci-openshift-hypershift-release-4.16-periodics-e2e-aws-ovn-conformance-serial](https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift_release/51393/rehearse-51393-periodic-ci-openshift-hypershift-release-4.16-periodics-e2e-aws-ovn-conformance-serial/1784507828677906432): ok (step phase pre & post passed)
[periodic-ci-openshift-hypershift-release-4.15-periodics-e2e-aws-ovn-conformance](https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift_release/51393/rehearse-51393-periodic-ci-openshift-hypershift-release-4.15-periodics-e2e-aws-ovn-conformance/1784507828610797568): ok

File: ci-operator/step-registry/cucushift/hypershift-extended/metadata/cucushift-hypershift-extended-metadata-ref.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+ref:
+  as: cucushift-hypershift-extended-metadata
+  from_image:
```
Comment (jianlinliu) [inline ci-operator/step-registry/cucushift/hypershift-extended/metadata/cucushift-hypershift-extended-metadata-ref.yaml:3]: how about using `from: cli` to be more clean.

File: ci-operator/step-registry/proxy/config-remove/proxy-config-remove-ref.yaml [hunk 2]
```diff
@@ -0,0 +1,15 @@
+ref:
+  as: proxy-config-remove
+  from_image:
```
Comment (jianlinliu) [inline ci-operator/step-registry/proxy/config-remove/proxy-config-remove-ref.yaml:3]: suggest to use `cli` to be more clean.

Comment (fxierh) [reply ci-operator/step-registry/proxy/config-remove/proxy-config-remove-ref.yaml]: Updated.

File: ci-operator/step-registry/cucushift/hypershift-extended/metadata/cucushift-hypershift-extended-metadata-ref.yaml (hunk 1 above)
Comment (jianlinliu) [reply ci-operator/step-registry/cucushift/hypershift-extended/metadata/cucushift-hypershift-extended-metadata-ref.yaml:3]: ah. got it.

File: ci-operator/step-registry/cucushift/hypershift-extended/install-private/cucushift-hypershift-extended-install-private-chain.yaml [hunk 3]
```diff
@@ -3,6 +3,9 @@ chain:
   steps:
   - ref: cucushift-hypershift-extended-install-private-config
   - ref: cucushift-hypershift-extended-install-private
+  env:
+  - name: HYPERSHIFT_AWS_REGION
+    default: ""
```
Comment (LiangquanLi930) [inline ci-operator/step-registry/cucushift/hypershift-extended/install-private/cucushift-hypershift-extended-install-private-chain.yaml:8]: We can remove this, step: cucushift-hypershift-extended-install-private-config, `HYPERSHIFT_AWS_REGION` has already been defined.

Comment (fxierh) [reply ci-operator/step-registry/cucushift/hypershift-extended/install-private/cucushift-hypershift-extended-install-private-chain.yaml]: Done.

**Comment by fxierh** [comment]
/assign @liangxia 

Could you please take a look ? Thanks.

**Comment by fxierh** [comment]
Had to rebase due to #51645.

**Pull request merged**